commitrmnocb/outcome.go

```diff
@@ -0,0 +1,325 @@
+) map[cciptypes.ChainSelector]cciptypes.SeqNum {
+	consensus := make(map[cciptypes.ChainSelector]cciptypes.SeqNum)
+
+	for chain, onRampMaxSeqNums := range onRampMaxSeqNumsByChain {
+		if f, exists := fChains[chain]; exists {
+			if len(onRampMaxSeqNums) < 2*f+1 {
+				// TODO: metrics
+				p.lggr.Warnf("could not reach consensus on onRampMaxSeqNums for chain %d "+
+					"because we did not receive more than 2f+1 observed sequence numbers, 2f+1: %d, "+
+					"len(onRampMaxSeqNums): %d, onRampMaxSeqNums: %v",
+					chain, 2*f+1, len(onRampMaxSeqNums), onRampMaxSeqNums)
+			} else {
+				sort.Slice(onRampMaxSeqNums, func(i, j int) bool { return onRampMaxSeqNums[i] < onRampMaxSeqNums[j] })
+				consensus[chain] = onRampMaxSeqNums[f]
+			}
+		} else {
+			// TODO: metrics
+			p.lggr.Warnf("could not reach consensus on onRampMaxSeqNums for chain %d "+
+				"because there was no consensus f value for this chain", chain)
+		}
+	}
+
+	return consensus
+}
+
+// Given a mapping from chains to a list of max seq nums on the OffRamp, return a mapping from chains
+// to a single max seq num. The consensus max seq num for a given chain is the max seq num with the most observations
+// that was observed at least f times, where f is the FChain of the dest chain.
+func (p *Plugin) offRampMaxSeqNumsConsensus(
+	offRampMaxSeqNumsByChain map[cciptypes.ChainSelector][]cciptypes.SeqNum,
+	fDestChain int,
+) map[cciptypes.ChainSelector]cciptypes.SeqNum {
+	consensus := make(map[cciptypes.ChainSelector]cciptypes.SeqNum)
+
+	for chain, offRampMaxSeqNums := range offRampMaxSeqNumsByChain {
+		seqNum, count := mostFrequentElem(offRampMaxSeqNums)
+		if count <= fDestChain {
+			// TODO: metrics
+			p.lggr.Warnf("could not reach consensus on offRampMaxSeqNums for chain %d "+
+				"because we did not receive a sequence number that was observed by at least f (%d) oracles, "+
+				"offRampMaxSeqNums: %v", chain, fDestChain, offRampMaxSeqNums)
+		} else {
+			consensus[chain] = seqNum
+		}
+	}
+
+	return consensus
+}
+
+// Given a mapping from chains to a list of FChain values for each chain, return a mapping from chains
+// to a single FChain. The consensus FChain for a given chain is the FChain with the most observations
+// that was observed at least f times, where f is the F of the DON (p.reportingCfg.F).
+func (p *Plugin) fChainConsensus(fChainValues map[cciptypes.ChainSelector][]int) map[cciptypes.ChainSelector]int {
+	consensus := make(map[cciptypes.ChainSelector]int)
+
+	for chain, fValues := range fChainValues {
+		f, _ := mostFrequentElem(fValues)
+		// TODO: uncomment when p.reportingCfg is added back
+		//if count < p.reportingCfg.F {
+		//	// TODO: metrics
+		//	p.lggr.Warnf("failed to reach consensus on fChain values for chain %d because no single f "+
+		//		"value was observed more than the expected %d times, found f value %d observed by only %d oracles, "+
+		//		"f values: %v",
+		//		chain, p.reportingCfg.F, f, count, fValues)
+		//}
```

Comment:
Lets add this back, was there an issue with passing through reportingPluginConfig?